src/components/Button/src/index.tsx

```diff
@@ -10,7 +10,7 @@ export interface ButtonProps extends BaseProps {
   /**
    * Simple click handler
    */
-  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>) => void;
```

Comment:
Instead of using an OR-operand here for every possible event, to include as many events as possible, I would use React's `SyntheticEvent` here. This is a wrapper around pretty much any and all native browser events (which you can obviously restrict by specifying it as a param for the `onClick` property.

Reply:
`MouseEvent` and `TouchEvent` are more specific than `SyntheticEvent` and include more specific fields which the user can use.